numba/np/arraymath.py

```diff
@@ -4251,3 +4251,32 @@ def impl(a, b):
         return _cross2d_operation(a_, b_)
 
     return impl
+
+
+@overload(np.trim_zeros)
+def np_trim_zeros(a, trim='fb'):
+    if not isinstance(a, (types.Array, types.Sequence, types.Tuple)):
+        raise TypeError('The first argument must be an array-like')
+    if hasattr(a, 'ndim') and a.ndim > 1:
+        raise TypingError('array must be 1D')
```

Comment:
Should we check that `trim` is a string or string literal?

Reply:
It seems like `np.trim_zeros(arr)` `np.trim_zeros(arr, 'f')` and `np.trim_zeros(arr, trim)` all produces expected output. 
Would we still need to add a check?

Reply:
It would improve the error message if someone does something like `np.trim_zeros(arr, 1)`.

Reply:
Oh, I see. I will add the check